sql/core/src/test/scala/org/apache/spark/sql/IntegratedUDFTestUtils.scala

```diff
@@ -191,6 +191,59 @@ object IntegratedUDFTestUtils extends SQLHelper {
     throw new RuntimeException(s"Python executable [$pythonExec] and/or pyspark are unavailable.")
   }
 
+  private def createPythonUDTF(funcName: String, pythonScript: String): Array[Byte] = {
+    if (shouldTestPythonUDFs) {
+      var binaryPandasFunc: Array[Byte] = null
+      withTempPath { codePath =>
+        Files.write(codePath.toPath, pythonScript.getBytes(StandardCharsets.UTF_8))
+        withTempPath { path =>
+          Process(
+            Seq(
+              pythonExec,
+              "-c",
+              "from pyspark.serializers import CloudPickleSerializer; " +
+                s"f = open('$path', 'wb');" +
+                s"exec(open('$codePath', 'r').read());" +
+                "f.write(CloudPickleSerializer().dumps(" +
+                s"($funcName, returnType)))"),
+            None,
+            "PYTHONPATH" -> s"$pysparkPythonPath:$pythonPath").!!
+          binaryPandasFunc = Files.readAllBytes(path.toPath)
+        }
+      }
+      assert(binaryPandasFunc != null)
+      binaryPandasFunc
+    } else {
+      throw new RuntimeException(s"Python executable [$pythonExec] and/or pyspark are unavailable.")
+    }
+  }
+
+  private lazy val pythonTableFunc: Array[Byte] = {
+    val script =
+      """
+        |from pyspark.sql.types import StructType, StructField, IntegerType
+        |returnType = StructType([
+        |  StructField("a", IntegerType()),
+        |  StructField("b", IntegerType()),
+        |  StructField("c", IntegerType()),
+        |])
+        |class SimpleUDTF:
+        |    def __init__(self):
+        |        self._count = 0
+        |
+        |    def eval(self, a: int, b: int):
+        |        self._count += 1
+        |        yield a, b, a + b
+        |        yield a, b, a - b
+        |        yield a, b, b - a
+        |
+        |    def terminate(self):
+        |        self._count = 0
```

Comment:
do we plan to reuse the class instance between multiple evaluations of input partition row sets? I would imagine it would be simpler for the user to not do so, therefore we do not need to think about resetting logic like this, and we can just assume that the __init__ method will freshly initialize and assign any class members as needed?